c7n/resources/s3.py

```diff
@@ -2498,15 +2498,15 @@ class Lifecycle(BucketActionBase):
                                 'And': {
                                     'type': 'object',
                                     'additionalProperties': False,
-                                    'items': {
+                                    'properties': {
```

Comment:
so i'm a bit surprised that this got through unit test runs, ci runs with schema validation on for unit test executed policies.

Reply:
looks like it passed because the lifecycle options with problematic schemas weren't being tested.

Reply:
@kapilt - there are a ton of options to the lifecycle and they can't all be specified together in a valid policy so I am not exercising every option.  I could have one fake test just for validating the schema that tries to utilize every single option and just validates the policy instead of running it.  Would that be useful?